File: Selection_Sort/SelectionSort.c
//C program to implement Selection Sort
 
  #include <stdio.h>
  #include <conio.h>
  
       //function to swap elements
  
		void swap(int *ele1, int *ele2)
		{
			int temp = *ele1;
			*ele1 = *ele2;
			*ele2 = temp;
		}
		
		//function to perform selection sort
		
		void SelectionSort(int arr[], int len)
		{
			int i, j, min;
			
			//finding minimum element in unsorted array
			
			  for(i = 0; i < len-1; i++)
			  {
			  	min = i;
			  	
			  	 for(j = i+1; j < len; j++)
			  	 {
			  	 	if(arr[j] < arr[min])
			  	 	   min = j;
			  	 	   
				   }
				 
				 //swapping minimum element with first element
				 
				 swap(&arr[min], &arr[i]);
				   
			  }
		}
		
		//function to display elements of an array
		
		void displayArray(int arr[], int size) 
         { 
         
            for (int i=0; i < size; i++) 
              printf("%d ", arr[i]); 
              
        printf("\n"); 
        } 
        
        //main function to control program execution by calling other functions
        
        int main()
        {
        	int length;
        	
        	printf("Enter the length of the array ");
        	  scanf("%d",&length);
		
        	int ar[length];
		
        	printf("Enter the elements in the array ");
			   
			   for(int i = 0; i < length; i++)
			       scanf("%d",&ar[i]);
				   
				
			SelectionSort(ar, length); 
			
             printf("Sorted array: \n"); 
             
                displayArray(ar, length); 
                
           return 0; 	     
		}

File: Selection_Sort/selection_sort.java
import java.util.Arrays;
import java.util.Scanner;

class SelectionSort {
	private static Scanner sc;

	public static void main(String args[]) {
		sc = new Scanner(System.in);

		System.out.println("Enter the number of elements");
		int n = sc.nextInt();

		System.out.println("Enter the elements");
		int arr[] = new int[n];
		for (int i = 0; i < n; i++)
			arr[i] = sc.nextInt();


		sort(arr);

		System.out.println("The sorted array is:");
		System.out.println(Arrays.toString(arr));
	}

	static void sort(int arr[]) {

		for (int i = 0; i < arr.length - 1; i++) {
			
			int k = i;
			for (int j = i + 1; j < arr.length; j++)
				if (arr[j] < arr[k])
					k = j;
			if (i != k) {
				int temp = arr[i];
				arr[i] = arr[k];
				arr[k] = temp;
			}
		}
	}
}